frontend/components/product-list/sections/FilterableProductsSection/SearchInput.tsx

```diff
@@ -1,76 +1,88 @@
 import {
-   chakra,
+   forwardRef,
    Icon,
    IconButton,
    Input,
    InputGroup,
+   InputGroupProps,
    InputLeftElement,
    InputRightElement,
 } from '@chakra-ui/react';
-import { useSearch } from '@lib/algolia';
+import { useDebounce } from '@ifixit/ui';
 import * as React from 'react';
 import { HiXCircle } from 'react-icons/hi';
 import { RiSearchLine } from 'react-icons/ri';
+import { useSearchBox } from 'react-instantsearch-hooks-web';
 
-interface SearchInputProps {
-   className?: string;
-}
+type SearchInputProps = InputGroupProps;
 
 const MAX_SEARCH_QUERY_LENGTH = 100;
+const DEBOUNCE_INTERVAL_MILLIS = 300;
 
-export const SearchInput = chakra(({ className }: SearchInputProps) => {
+export const SearchInput = forwardRef<SearchInputProps, 'div'>((props, ref) => {
+   const searchBox = useSearchBox();
    const inputRef = React.useRef<HTMLInputElement>(null);
-   const [query, search] = useSearch();
+   const [searchQuery, setSearchQuery] = React.useState(searchBox.query);
+   const debouncedQuery = useDebounce(searchQuery, DEBOUNCE_INTERVAL_MILLIS);
+   const lastDeboucedQueryRef = React.useRef<string | null>(null);
+
+   React.useEffect(() => {
+      if (debouncedQuery !== searchBox.query) {
+         if (lastDeboucedQueryRef.current === debouncedQuery) {
+            setSearchQuery(searchBox.query);
```

Comment:
I'm struggling to understand this part... where we update the input box to the value from algolia. In other cases the updates go the other way (inputbox -> algolia).

Reply:
I've refactored the code so hopefully is more readable. Basically it's decoupling input state from Algolia context state so the input feels more responsive.